src/lib/common/port_monitor.c

```diff
@@ -0,0 +1,107 @@
+#include "port_monitor.h"
+#include "bm_config.h"
+#include "bm_os.h"
+#include "configuration.h"
+#include "device.h"
+#include "l2.h"
+#include "spotter.h"
+#include <inttypes.h>
+#include <string.h>
+
+#define port_monitor_time_ms_key "disableUnusedPortsTimeMs"
+#define port_monitor_time_ms_default (0U)
+#define port_monitor_log "port_monitor.log"
+
+static uint16_t ENABLED_PORT_MASK = 0;
+
+/*!
+  @brief Link change callback to set ports which have come online
+
+  @details Will set a masked value of the ports which have come online.
+
+  @param port port that has had a link change
+  @param state state of the port
+ */
+static void port_monitor_link_change(uint8_t port, bool state) {
+  // If the port is online, update the mask
+  if (state) {
+    ENABLED_PORT_MASK |= 1 << (port - 1);
+  }
+}
+
+/*!
+  @brief Timer callback handler to determine if all ports have come online
+
+  @details Checks if all ports on the node have come online, if a port is
+           not online, disable the unused port.
+ 
+  @param timer unused
+ */
+static void port_monitor_timer_handler(BmTimer timer) {
+  (void)timer;
+  uint8_t ports = bm_l2_get_port_count();
+
+  for (uint8_t i = 0; i < ports; i++) {
+    uint16_t enabled = (1 << i) & ENABLED_PORT_MASK;
+    uint8_t port_num = i + 1;
+    if (enabled == 0) {
+      if (port_num != 1) {
+        const char *fmt_disable_str =
+            "Disabling network port %d on node: %" PRIx64 ", it never came online\n";
+        spotter_log(0, port_monitor_log, USE_TIMESTAMP, fmt_disable_str, port_num);
+        spotter_log_console(0, fmt_disable_str, node_id(), port_num);
+        bm_l2_netif_enable_disable_port(port_num, false);
+      } else {
+        const char *fmt_err_str = "Could not disable network port %d on node: %" PRIx64
```

Comment:
Please add a comment with a brief explanation including the page number in the datasheet. Future folks will be very confused looking at this code and wondering why we compare to magic number 1.